eth/db/chain.py

```diff
@@ -395,6 +396,28 @@ def get_transaction_index(self, transaction_hash: Hash32) -> Tuple[BlockNumber,
         transaction_key = rlp.decode(encoded_key, sedes=TransactionKey)
         return (transaction_key.block_number, transaction_key.index)
 
+    def get_receipt_by_index(self,
+                             block_number: BlockNumber,
+                             receipt_index: int,
+                             receipt_class: Type[Receipt]) -> Receipt:
```

Comment:
Is there a reason this needs to be provided by the user?  Currently all vms use the same `Receipt` class.

Reply:
@pipermerriam I just put that up, so as to mimic the `get_transaction_by_index` function. Guess I will add a default value which is `Receipt` to this function.